File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
from typing import List

class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        n = len(nums)
        l, r = 0, 0

        total = sum(nums)
        if total < target:
            return 0

        curr_sum = 0
        ans = float('inf')

        while r < n:
            curr_sum += nums[r]

            while curr_sum >= target:
                ans = min(ans, r - l + 1)
                curr_sum -= nums[l]
                l += 1

            r += 1

        return ans
            
        

File: Data Structures & Algorithms/jump-game-vii/submission-0.cpp
class Solution {
public:
    bool canReach(string s, int minJump, int maxJump) {
        int l=0,r=0;
        int n=s.size();
        vector<bool> dp(n,false);
        if(s[0]=='0'){
            dp[0]=true;
        }else{
            return false;
        }
        while(l<n){
            if(!dp[l]){
                l++;
                continue;
            }
            for(int i=l+minJump;i<=min(l + maxJump,n- 1);i++){
                dp[i] = (s[i] == '0') ? true : dp[i];   
            }
            l+=minJump;
        }
        return dp[n-1];

        
    }
};